### ISSUE TITLE
multiline input: unable to delete previous lines

### ISSUE THREAD

**Issue by mcpcpc** [andir/ate#22]
This behavior was first reported as part of https://github.com/mcpcpc/kirc/issues/39 and appears to impact multiple terminal emulators (e.g. foot and st) while behaving fine in others (e.g. xterm)

To reproduce, one simply needs to read input from stdin (e.g. fgets) and type more characters than the width of the terminal window.  the text that is longer than window width is then printed as a new line, making the previous line uneditable. 

A simple verifiable solution (written in C) includes the following:

```
#include <stdio.h>
#include <stdlib.h>

int
main() {
    char a[1000];
    fgets(a, 1000, stdin);
    printf("string: %s\n", a);
    return 0;
}
```

**Comment by andir** [comment]
Yeah, I can confirm that this is happening. In this case It is probably worth testing with gnome-terminal and/or reporting the issue to with libvte as that is basically all I'm using here.

**Comment by mcpcpc** [comment]
submitted upstream to libvte: https://gitlab.gnome.org/GNOME/vte/-/issues/287

**Comment by andir** [comment]
This could also be somewhat related to the issue: https://gitlab.gnome.org/GNOME/vte/-/issues/62

**Comment by andir** [comment]
I am always within tmux and/or neovims terminal thus I haven't really
noticed this.

**Comment by mcpcpc** [comment]
Here's another conversation I am having with the developer of `foot`: [link](https://codeberg.org/dnkl/foot/issues/150).
